RZAssemblage/RZProxyAssemblage.m

```diff
@@ -139,9 +120,13 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
             }];
         }
         else if ( changeType == NSKeyValueChangeRemoval ) {
-            [indexes enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL *stop) {
-                [self.changeSet removeAtIndexPath:[NSIndexPath indexPathWithIndex:idx]];
-            }];
+            NSArray *oldValues = change[NSKeyValueChangeOldKey];
+            for (NSUInteger i = 0, index = [indexes firstIndex];
+                 i < indexes.count;
+                 i++, index = [indexes indexGreaterThanIndex:index] ) {
```

Comment:
hah, yea I was pushing it. This is why I try to wait till the next day to merge.